calico/felix/fsocket.py

```diff
@@ -60,16 +62,52 @@ class Socket(object):
     def __init__(self, type, config):
         self.config = config
         self.type = type
-        self.remote_addr = None
-        self.port = Socket.PORT[type]
+        self._remote_addr = None
+        self._port = Socket.PORT[type]
         self._zmq = None
-        self.last_activity = None
-        self.request_outstanding = False
+        self._last_activity = None
+
+        #: _request_outstanding is always FALSE for not REQUEST_TYPES
+        self._request_outstanding = False
 
         if type in Socket.EP_TYPES:
-            self.remote_addr = self.config.PLUGIN_ADDR
+            self._remote_addr = self.config.PLUGIN_ADDR
+        else:
+            self._remote_addr = self.config.ACL_ADDR
+
+        if type in Socket.REQUEST_TYPES:
+            self._send_queue = collections.deque()
+        else:
+            self._send_queue = None
+
+        self._subscriptions = set()
+
+        if type == Socket.TYPE_EP_REQ:
+            self.descr = ("plugin %s:%s (request connection %s)" %
+                          (self._remote_addr, self._port, self.type))
+        elif type == Socket.TYPE_EP_REP:
+            self.descr = ("plugin (bound socket %s:%s, %s)" %
+                          (self.config.LOCAL_ADDR, self._port, self.type))
+        elif type == Socket.TYPE_ACL_REQ:
+            self.descr = ("ACL Manager %s:%s (request connection %s)" %
+                          (self._remote_addr, self._port, self.type))
         else:
-            self.remote_addr = self.config.ACL_ADDR
+            self.descr = ("ACL Manager %s:%s (subscription connection %s)" %
+                          (self._remote_addr, self._port, self.type))
```

Comment:
This `else` is asking for bugs in future. I know I did it when I first wrote the code, but we should remove it now: replace with `elif` and have the `else` throw an exception.